### ISSUE TITLE
Steam VR cant enable direct mode since 1.16.10 [BUG]

### ISSUE THREAD

**Issue by jonbitzen** [ValveSoftware/SteamVR-for-Linux#233]
Since 1.16.10, whenever I start my headset, I get the SteamVR compositor in the desktop, but the HMD itself is black. The SteamVR popup asks me to enable Direct Mode, I accept, it restarts, and we are still stuck with the compositor on the desktop and the HMD black, except no sign of the SteamVR popup anymore.

I tried the latest beta (1.7.3) it has the same issues.

Im running LinuxMint 19, GTX 1060/6Gb, latest Nvidia drivers (430.40).

I've tried going back to previous nVidia drivers (415.27) and that didn't help at all either.

**To Reproduce**
1. Start SteamVR with the HMD connected to the PC
2.  The SteamVR compositor shows up on the Desktop, the HMD display is blank.
3.  The SteamVR popup control says direct mode is not enabled; it offers to enable it and restart SteamVR.  Click accept.
4.  SteamVR popup exits, the SteamVR compositor on the desktop goes away.
5.  SteamVR compositor comes back up on the desktop, the control popup never comes back.

**Expected behavior**
The HMD display should start up, the desktop compositor should go away.

Using SteamVR 1.7.3 beta

System Report Below
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
    CPU Family:  0x6
    CPU Model:  0x3c
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  4000 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Linux Mint 19 Tara (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.15.0-55-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11906000
    X Window Manager:  Mutter (Muffin)
    Steam Runtime Version:  steam-runtime_0.20190711.3

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1060 6GB/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 430.40
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 30 Hz
    VendorID:  0x10de
    DeviceID:  0x1c03
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  3840 x 2160
    Desktop Resolution: 3840 x 2160
    Primary Display Size: 73.70" x 41.46" (84.53" diag)
                                            187.2cm x 105.3cm (214.7cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 6144 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC892

Memory:
    RAM:  7904 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  119674 Mb
    Largest Free Hard Disk Block:  79159 Mb
    VR Headset: HTC Vive detected

Recent Failure Reports:

**Comment by lostgoat** [comment]
Hi @jonbitzen ,

Thanks for opening the issue here.

Can I trouble you to paste a SteamVR System Report to a gist? The instructions are here:
https://github.com/ValveSoftware/SteamVR-for-Linux/blob/master/CapturingSystemReport.md

Since you are having display related troubles, seeing the output of `xrandr --verbose` might also help.

**Comment by Sevdrop** [comment]
I had that too. Reinstalling SteamVR solved the issue..

**Comment by jonbitzen** [comment]
Heres the gist link with my system report for this issue.  I'll get the xrandr info too soon.

https://gist.github.com/jonbitzen/332661bc5e42002cbb9e26e3e1a240dc

**Comment by jonbitzen** [comment]
Heres the xrandr report:

https://gist.github.com/jonbitzen/d8d47f98dab069b05f7fc1a934be1d78

**Comment by lostgoat** [comment]
Hey jonbitzen,

It seems like vrcompositor can't find your HMD display. Can you double check the display port connection to the HMD and to the PC?

**Comment by lostgoat** [comment]
Sorry, two more things. Try unplugging the HMD for the power connector for 30 seconds to let it fully power cycle.

Second, to make testing easier, you don't need to relaunch all of steamvr. You can just run xrandr (without the verbose option) and you should be able to see your HMD with the correct modes.

**Comment by jonbitzen** [comment]
Hey thanks it’s probably safe to close this.

I think it’s a bad cable.  I changed the hmd over from DisplayPort to hdmi, and switched my monitor to a different input and everything worked.  Maybe the DisplayPort cable got damaged, I recently took the entertainment center apart twice, so anything could have happened.  I had it all apart a few days and it straddled the SteamVR update so I guess correlation isn’t causation after all.  Science wins again!

Thanks,

jonbitzen
